submissions/yhorodechna/side-menu-dom/js/script.js

```diff
@@ -0,0 +1,107 @@
+let _navEl;
+const getNavElement = () => {
+    if (!_navEl) {
+        _navEl = document.querySelector('#nav');
+    };
+    return _navEl;
+};
+let ACTIVE_EL_ID;
+
+function findHeaderLinkOrNavLink(clickedElement) {
+    let res;
+    let current = clickedElement;
+    while (!res && current) {
+        const { className } = current;
+        if (className) {
+            if (className.includes('nav__link')) {
+                res = {
+                    navLink: current
+                }
+                continue;
+            }
+            else if (className.includes('header__link')) {
+                res = {
+                    headerLink: current
+                }
+                continue;
+            }
+        }
+        current = current.parentNode;
+    }
+    return res;
+};
+
+function handleContainerClick({ target }) {
+    const { navLink, headerLink } = findHeaderLinkOrNavLink(target);
+
+    if (navLink) {
+        if (ACTIVE_EL_ID !== navLink.id) {
+            if (ACTIVE_EL_ID) {
+                document.querySelector('#' + ACTIVE_EL_ID).classList.remove('active');
+            };
+            ACTIVE_EL_ID = navLink.id;
+            navLink.classList.add('active');
+            showArticle(navLink);
+        };
+        getNavElement().classList.remove('nav__full');
+    } else if (headerLink) {
+        getNavElement().classList.add('nav__full');
+    };
+};
+
+function createNavList({ items, parentEl, defaultId }) {
+    ACTIVE_EL_ID = defaultId;
+    findHeaderLinkOrNavLink();
+
+    const navElements = items.map((item) =>
+        `<li class="nav__item">
+            <a id="${item.id}" href="#" class="nav__link ${defaultId === item.id ? 'active' : ''}">
+                <span class="nav__text">${item.title}</span> 
+            </a>
+        </li>`
+    );
+    const html = `
```

Comment:
please, try not to use such naming, it is too general. At least`navHtml`

Reply:
Changed.